funcframework/framework.go

```diff
@@ -99,33 +117,38 @@ func registerEventFunction(path string, fn interface{}, h *http.ServeMux) {
 	})
 }
 
+func registerCloudEventFunction(path string, fn func(cloudevents.Event), h *http.ServeMux) {
+	p, err := cloudevents.NewHTTP()
+	if err != nil {
+		log.Fatalf("failed to create protocol: %s", err.Error())
+	}
+
+	handleFn, err := cloudevents.NewHTTPReceiveHandler(context.Background(), p, fn)
```

Comment:
What is this context used for? Should we pass it as an argument?

Reply:
Possibly cancellation? I'm not positive. Adding it would make the function signature func(context.Context, cloudevents.Event). Is that acceptable? @grant

Reply:
I think if this is something expected by `cloudevents`, we should pass it through. It would probably come from `r.Context()` at some point.

Reply:
@grant do you have an opinion on the function signature? I'm inclined to agree with Tyler, despite the departure from the contract (particularly given that the .NET framework has as similar departure, with additional parameters that are language-idiomatic).

Reply:
Yes this makes sense (ctx, ce).

Example function:
https://knative.dev/docs/eventing/samples/helloworld/helloworld-go/#recreating-the-sample-code

Reply:
Use `ctx` here?

Reply:
Where?

Reply:
it is used for shutdown and lifecycle management of the process.

Reply:
we also use it to pass in options to overload special things.

Reply:
It feels like we should plumb a `context.Context` through as an argument from the user instead of using `context.Background()`.

(Weird -- this was still a pending comment from a couple days ago, even through other comments went through. Sorry!)